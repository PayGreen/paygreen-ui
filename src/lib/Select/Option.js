import React from 'react';
import PropTypes from 'prop-types';

const Option = props => {
    const { option, readOnly, defaultValue } = props;

    return (
        <option
            value={option.value}
            disabled={
                option.disabled || (readOnly && option.value !== defaultValue)
            }
        >
            {option.text}
        </option>
    );
};

Option.propTypes = {
    option: PropTypes.shape({
        value: PropTypes.string.isRequired,
        text: PropTypes.string.isRequired,
        disabled: PropTypes.bool,
    }).isRequired,
    readOnly: PropTypes.bool,
};

Option.defaultProps = {
    readOnly: false,
};

export default Option;
