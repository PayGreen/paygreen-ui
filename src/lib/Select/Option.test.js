import React from 'react';
import TestRenderer from 'react-test-renderer';
import Option from './Option';

const sampleOption = {
    value: 'first',
    text: 'First option',
};

it('renders without crashing', () => {
    const optionBlock = TestRenderer.create(
        <Option option={sampleOption} />,
    );
    expect(optionBlock.toJSON()).toMatchSnapshot();
});
