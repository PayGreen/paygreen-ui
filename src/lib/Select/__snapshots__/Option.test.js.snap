// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<option
  disabled={false}
  value="first"
>
  First option
</option>
`;
